### PR TITLE
Stop trying to create ByRef-like Dummies on TMFs lower than netstandard2.1

### PR DESCRIPTION
**Pull request by blairconrad**
As with #2019, an incomplete pull request at this time. Avoids touching the CreationResult code at all, as you suggested, @thomaslevesque.

### CODE REVIEW

**Comment by blairconrad** [comment]
@thomaslevesque, the last commit is a candidate for squashing into the previous, I think. (Under any PR.)

File: src/FakeItEasy/TypeExtensions.cs [hunk 1]
```diff
@@ -31,6 +32,15 @@ public static bool IsNullable(this Type type)
                 || (type.IsGenericType && type.GetGenericTypeDefinition() == typeof(Nullable<>));
         }
 
+        public static bool IsByRefLike(this Type type)
+        {
+#if LACKS_ISBYREFLIKE
+            return type.GetCustomAttributesData().Any(att => att.AttributeType.FullName == "System.Runtime.CompilerServices.IsByRefLikeAttribute");
```
Comment (thomaslevesque) [inline src/FakeItEasy/TypeExtensions.cs:38]: Well, that wasn't so bad after all. As I said in the other PR, I don't think we really _need_ to handle this case for netstandard2.0, but if it's no harder than that, I guess it doesn't hurt. We can always drop this method when we drop netstandard TFMs.

**Review by thomaslevesque — COMMENTED** [review]
I think this is my favorite version. Straightforward, no major changes, no big-bang refactoring of `CreationResult`... What's not to like?

File: src/FakeItEasy/TypeExtensions.cs (hunk 1 above)
Comment (blairconrad) [reply src/FakeItEasy/TypeExtensions.cs:38]: Yeah, it ended up being simpler than I thought.

**Review by thomaslevesque — APPROVED** [review]
Thanks, @blairconrad!

**Comment by blairconrad** [comment]
Thanks, @thomaslevesque. As mentioned in sidechat, my commit message was _terrible_. Redoing in #2026.

**Pull request merged**